### ISSUE TITLE
Windows security detects app as threat. Installation has failed.

### ISSUE THREAD

**Issue by huuxfluux** [ffxiv-teamcraft/ffxiv-teamcraft#2942]
Windows security detects app as threat and installation fail

Downloaded `ffxiv-teamcraft-Setup-11.3.14.exe` and started the install.
First i had to confirm that i trusted the executable.
Then my installation failed.

Note: I don't have any antivirus beides Windows Security.

**To Reproduce**
1. Click install 'ffxiv-teamcraft-Setup-11.3.14.exe' (downloaded from github)
3. Click on 'advanced options'
4. Click on 'run anyway'
6. See error
![image](https://github.com/user-attachments/assets/1e4a95b7-1299-475b-9388-5ffc71a1a495)

**Screenshots**
![image](https://github.com/user-attachments/assets/51c5d6ee-f3d9-4ca9-8438-d7c3eb0a0075)

**Desktop (please complete the following information):**
![Screenshot 2024-11-06 112621](https://github.com/user-attachments/assets/67297baf-acb5-42f6-991d-222ad1c8bec0)

**Additional context**
Logfile from the install error:
[SquirrelSetup.log](https://github.com/user-attachments/files/17645514/SquirrelSetup.log)

**Comment by Officerfriendly1** [comment]
Need to go into Windows Defender and check the action history to find the pack being quarantined. The current setup with Defender will take getting the pack whitelisted and the injector for packet watching.

**Comment by AxelTerizaki** [comment]
Can confirm it happens with 11.3.15 as well.

**Comment by Supamiu** [comment]
I'm on it, trying various tricks to solve that but be sure that it's a false positive, nothing has changed except AVs suddently going mad at us.

**Comment by AxelTerizaki** [comment]
I guessed. I have run into this kind of issue with my own software as well. Good luck :)

**Comment by Tabcaps54** [comment]
Yeah I'm getting this error from Chrome as well... it also uninstalled my old version of Teamcraft so now I have nothing lol 

![image](https://github.com/user-attachments/assets/a85fe4ff-c9e9-4860-aa18-7ca909d0c213)

**Comment by Khelendil** [comment]
Same on Windows 10 pro but the Name is slightly differnt.
Used the exe installer n it is called B!ml instead of H!ml 

Attached the Logs from the Setup, hope they might help.

[Squirrel-Install.log](https://github.com/user-attachments/files/17727440/Squirrel-Install.log)
[SquirrelSetup.log](https://github.com/user-attachments/files/17727442/SquirrelSetup.log)

**Comment by Fijvect** [comment]
same issue! :(

**Comment by m13253** [comment]
As of today, most anti-virus software (65 out of 66) no longer mark this file as a threat.
https://www.virustotal.com/gui/file/0f78857c01d08e7b80e30c796bcce7b3006a10862d9f1d5188bb58b251a9776e

I think people in anti-virus companies have analyzed the file and determined it safe.

In my honest opinion, the best way to deal with false positives is not trying to workaround it in the source code.
Instead, submit these executables to an anti-virus company to let real human analyze the file.

Alternatively, I guess buying a digital signature would probably also reduce false positive rate.

**Comment by Supamiu** [comment]
Yeah we've submitted the files and an Open Source code signing cert purchase is ongoing with Certum which is extremely slow to deliver

**Comment by Khelendil** [comment]
Just installed latest Version 11.3.21 and Windows Defender complained its a Virus

![TC_11_3_21_-_Virus_Warning](https://github.com/user-attachments/assets/b619efbc-27a8-4b59-b6c5-8055d78f246c)

Installation went through and TC also started though.

**Comment by Lalalavicious** [comment]
Just so folks are aware, I've added [a section on the Troubleshooting page for the TC wiki](https://wiki.ffxivteamcraft.com/troubleshooting#current-issues-with-antivirus-false-positives) for steps to go through upon running into these issues! As far as I'm aware, that particular set of steps has resolved any and all of these kinds of problems people are now having. (Examples of behaviors you might run into: getting a warning like the above that Teamcraft has malware (not true, of course); getting "Installation has failed" or -- when updating from fairly old versions -- some kind of Javascript error message when trying to install; automatic updates get stuck, requiring manual download of the latest update; or, of course, if TC itself might say it detected that your antivirus/Windows Security/Windows Defender/etc. has eaten files required for packet capture, because that's a thing they like to do even when the install succeeds.)

There's still a measure or two that might help avoid some cases of this, but most apps like ours that run into this issue end up just giving up entirely and putting a caveat that your AV might block it... and I've really come to understand exactly why that is, haha. [Insert Yoshi-P's "NIGHTMARE" clip here]

**Comment by brianaydemir** [comment]
At least for me, as of v11.3.22, Windows Security is no longer flat out blocking installation. It's more of a, "Are you sure?"

![report](https://github.com/user-attachments/assets/32a82863-69c9-411b-9af7-17649d41631c)

One of the allowed actions for this alert is, "Allow on device," so there's no need to go mucking around with manual exclusions and the like.
